src/Microsoft.AspNet.Http.Features/IQueryCollection.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.Http
+{
+    /// <summary>
+    /// Represents the HttpRequest query string collection
+    /// </summary>
+    public interface IQueryCollection : IDictionary<string, StringValues>
+    {
+        /// <summary>
+        /// IQueryCollection has a different indexer contract than IDictionary, where it will return StringValues.Empty for missing entries.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns>The stored value, or StringValues.Empty if the key is not present.</returns>
+        new StringValues this[string key] { get; set; }
```

Comment:
I think this contract should only define `get` like IReadableStringCollection did. `set` raises a lot of consistency questions between request.Query and request.QueryString. The same goes for Form and Cookies.

I'm not positive we need the full IDictionary area either, but there's a lot more code to read here..